scipy/stats/kde.py

```diff
@@ -619,6 +625,132 @@ def logpdf(self, x):
                                       self._norm_factor)
 
         return result
+      
+    def pdf_marginal(self, x, axis):
+        """
+        Evaluate the estimated marginal pdf on a provided set of points.
+        
+        Parameters
+        ----------
+        points : (# of dimensions, # of points)-array
+            Alternatively, a (# of dimensions,) vector can be passed in and
+            treated as a single point.
+        axis : int or 1-d array-like
+            Axis (axes) along which marginal pdf is evaluated. In the case of
+            1-d array-like, the elements should also be integers.
+        
+        Returns
+        -------
+        values : (# of points,)-array
+            The values at each point.
+        
+        Raises
+        ------
+        ValueError : if the dimensionality of the input points is larger than
+                     the dimensionality of the KDE, or if it's different from
+                     the length of axis, or if the input axis is not consisted 
+                     of integers.
+        
+        """
+        points = atleast_2d(x)
+        ax = atleast_1d(axis)
+
+        if not issubclass(ax.dtype.type, numbers.Integral):
+            msg = "elements of axis should be intergers"
+            raise ValueError(msg)
+
+        d, m = points.shape
+        if ax.size > self.d or ax.size == 0:
+            msg = "axis have dimension %s, dataset has dimension %s" % \
+                (ax.size, self.d)
+            raise ValueError(msg)
+
+        if d != ax.size:
+            if d == 1 and m == ax.size:
+                # points was passed in as a row vector
+                points = reshape(points, (m, 1))
+                d = ax.size
+                m = 1
+            else:
+                msg = "points have dimension %s, axis has dimension %s" % \
+                (d, ax.size)
+
+        cov = self.covariance[ax[:, newaxis], ax]
+        inv_cov = linalg.inv(cov)
```

Comment:
The usual comment applies: never invert a matrix unless you just need the inverse.

Reply:
Got it.